src/com/amazon/ion/benchmark/WriteRandomIonValues.java

```diff
@@ -527,14 +528,42 @@ private static void writeDataToFile(IonType type, IonWriter writer, Random rando
                 writer.writeDecimal(WriteRandomIonValues.constructDecimal(expValRange, coefficientDigitRange));
                 break;
             case INT:
-                WriteRandomIonValues.constructInt(random, writer);
+                long intValue = WriteRandomIonValues.constructInt();
+                writer.writeInt(intValue);
                 break;
             case TIMESTAMP:
                 Timestamp timestamp = WriteRandomIonValues.constructTimestamp(timestampTemplate);
                 writer.writeTimestamp(timestamp);
                 break;
             case STRUCT:
-                WriteRandomIonValues.constructIonStruct(fields, annotations, writer);
+                if (annotationMap != null) {
+                    String annotationConstraint = annotationMap.get(IonSchemaUtilities.KEYWORD_CONSTRAINT).toString();
+                    IonList annotationList = (IonList) annotationMap.get(IonSchemaUtilities.KEYWORD_ANNOTATIONS);
+                    if (annotationConstraint.equals(IonSchemaUtilities.KEYWORD_REQUIRED)) {
+                        for (int i = 0; i < annotationList.size(); i++) {
+                            int index = random.nextInt(annotationMap.size());
+                            annotation = annotationList.get(index);
+                            WriteRandomIonValues.constructIonStruct(fields, annotation, writer);
```

Comment:
Is it acceptable for more than one annotation to be present on a single struct?

Reply:
> Is it acceptable for more than one annotation to be present on a single struct?

No it is not acceptable, only one annotation should be assigned to a single struct.

Reply:
Do you mean that Ion Schema doesn't allow users to specify that a struct may have multiple annotations?

Reply:
> Do you mean that Ion Schema doesn't allow users to specify that a struct may have multiple annotations?

I review the specification of Ion Schema found that there's no way to specify whether the value has one or multiple annotations in the schema annotations constraint, but both ways are acceptable by default(assign the generated value with only one or multiple annotations). However in the method above, only one annotation was assigned to each value. And I think the reorganization of logic of this part is necessary. Changes will not be included in the following commit but will be updated in the next PR.